core/drand_beacon_control.go

```diff
@@ -370,36 +370,43 @@ func (bp *BeaconProcess) runDKG(leader bool, group *key.Group, timeout uint32, r
 		Log:            bp.log,
 	}
 	phaser := bp.getPhaser(timeout)
-	board := newEchoBroadcast(bp.log, bp.version, beaconID, bp.privGateway.ProtocolClient,
-		bp.priv.Public.Address(), group.Nodes, func(p dkg.Packet) error {
-			return dkg.VerifyPacketSignature(config, p)
-		})
-	dkgProto, err := dkg.NewProtocol(config, board, phaser, true)
-	if err != nil {
-		return nil, err
-	}
 
 	bp.state.Lock()
 	dkgInfo := &dkgInfo{
 		target: group,
-		board:  board,
 		phaser: phaser,
 		conf:   config,
-		proto:  dkgProto,
+		// these can't be set yet, but at least _some_ info needs to be set for the echo broadcast
+		board: nil,
+		proto: nil,
 	}
 	bp.dkgInfo = dkgInfo
 	if leader {
+		// phaser will kick off the first phase for every other nodes so
+		// nodes will send their deals
+		bp.log.Infow("", "init_dkg", "START_DKG")
+		go phaser.Start()
+
 		bp.dkgInfo.started = true
 	}
 	metrics.DKGStateChange(metrics.DKGInProgress, beaconID, leader)
 	bp.state.Unlock()
 
-	if leader {
-		// phaser will kick off the first phase for every other nodes so
-		// nodes will send their deals
-		bp.log.Infow("", "init_dkg", "START_DKG")
-		go phaser.Start()
+	board := newEchoBroadcast(bp.log, bp.version, beaconID, bp.privGateway.ProtocolClient,
+		bp.priv.Public.Address(), group.Nodes, func(p dkg.Packet) error {
+			return dkg.VerifyPacketSignature(config, p)
+		})
+
+	dkgProto, err := dkg.NewProtocol(config, board, phaser, true)
```

Comment:
The reason why this change works is that `dkg.NewProtocol` is starting immediately in Kyber and uses the broadcast channel directly, so we need to make sure the nodes are ready to receive broadcasts, and for that `dkgInfo` must not be `nil`.